### PR TITLE
Enabling adjoint sensitivities in Tpetra build and Epetra cleanup

### PR DESCRIPTION
**Pull request by mperego**
This branch enables the adjoint sensitivities (depending on trilinos pull request trilinos/#3055) for the Tpetra build for scalar and distributed parameters.
It also removes a lot of Epetra code that after these changes is not used any longer, in particular in the responses folder that are now Epetra free. 
There are also a few random fixes in FELIX (sorry for not separating them from this pull request).

Before merging I'll wait until the trilinos pull request is merged into trilinos/develop. However this "should" compile with the current develop branch but some tests will fail.

### CODE REVIEW

**Comment by bartgol** [comment]
It's a big PR. Will take me a bit to review it. But it seems it will take a while anyways, since we are waiting on the trilinos PR to go in first...

Thanks for the effort!

**Comment by mperego** [comment]
I noticed that after rebasing my branch with master these test failed:
NSVortexShedding2D_TransIRK
NSVortexShedding2D_TransIRK_Tpetra
I would guess that these are caused by today commits by @ikalash and don't seem related with my changes.

**Comment by mperego** [comment]
Thanks @bartgol !  A lot of the changes are just code removals.. it might take less than you expect.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+PROJECT_ROOT = $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
```
Comment (bartgol) [inline Makefile]: What is this file? I don't see it in master.

File: src/FELIX/evaluators/FELIX_L2ProjectedBoundaryLaplacianResidual_Def.hpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ L2ProjectedBoundaryLaplacianResidual(Teuchos::ParameterList& p, const Teuchos::R
   w_side_measure     = decltype(w_side_measure)(w_side_measure_name, dl_side->qp_scalar);
   side_tangents      = decltype(side_tangents)(side_tangents_name, dl_side->qp_tensor_cd_sd);
   bdLaplacian_L2Projection_res = decltype(bdLaplacian_L2Projection_res)(residual_name, dl->node_scalar);
-  coordVec = decltype(coordVec)(coords_name, dl->qp_coords);
+  coordVec = decltype(coordVec)(coords_name, dl->vertices_vector);
```
Comment (bartgol) [inline src/FELIX/evaluators/FELIX_L2ProjectedBoundaryLaplacianResidual_Def.hpp]: This contradicts your commit message. I believe this is correct, but then, can you reword the commit?

**Comment by mperego** [comment]
Thanks @bartgol, I reworded the wrong commit message and removed the Makefile.
Yes, it was very satisfactory to removed all that code.
I'll merge as soon as the corresponding Trilinos changes are merged into master.

**Pull request merged**